[feature request] implent referrer rebates (binance, Spot and Future markets)

After my issue in https://github.com/provinzio/CoinTaxman/issues/58 i wanted to post some more lines from the whole transaction history. Please note i didn't trade futures (maybe someone else can help out with a history here). 

Referrer rebates line is not recognized under spot (2021-08-15 11:56:15,108 book         WARNING  Could not recognize operation `Referrer rebates` in  binance file `X:\CoinTaxman-main\account_statements\part-00000-154c5b63-ff2c-4615-9ed2-4d89319596e4-c000.csv:45475`.) and future markets (AssertionError: Other types than Spot are currently not supported. Please create an Issue or PR.) .

UTC_Time,Account,Operation,Coin,Change,Remark
2021-05-12 00:00:36,USDT-Futures,Referrer rebates,BNB,0.00000433,""
2021-05-12 00:01:57,Spot,Fee,BNB,-0.00025714,""
2021-05-12 00:01:57,Spot,Buy,RLC,24.00000000,""
2021-05-12 00:01:57,Spot,Transaction Related,BTC,-0.00404808,""
2021-05-12 00:03:07,USDT-Futures,Referrer rebates,BNB,5.1E-7,""
2021-05-12 00:04:05,USDT-Futures,Referrer rebates,BNB,4.3E-7,""
2021-05-12 00:04:21,Spot,Buy,DOTDOWN,89.14000000,""
2021-05-12 00:04:21,Spot,Fee,BNB,-0.00008958,""
2021-05-12 00:04:21,Spot,Transaction Related,USDT,-80.00315000,""
2021-05-12 00:05:07,USDT-Futures,Referrer rebates,BNB,2.1E-7,""
2021-05-12 00:06:06,USDT-Futures,Referrer rebates,BNB,0.00000122,""
2021-05-12 00:07:50,Spot,Fee,BNB,-0.00051858,""
2021-05-12 00:07:50,Spot,Buy,RLC,49.00000000,""
2021-05-12 00:07:50,Spot,Transaction Related,BTC,-0.00809725,""
2021-05-12 00:10:05,USDT-Futures,Referrer rebates,BNB,3.3E-7,""
2021-05-12 00:10:05,USDT-Futures,Referrer rebates,BNB,8.2E-7,""
2021-05-12 00:15:06,USDT-Futures,Referrer rebates,BNB,4.4E-7,""
2021-05-12 00:30:17,Spot,Fee,BNB,-0.00012729,""
2021-05-12 00:30:17,Spot,Buy,DASH,0.27000000,""
2021-05-12 00:30:17,Spot,Transaction Related,BTC,-0.00200259,""
2021-05-12 00:50:11,Spot,Fee,BNB,-0.00012843,""
2021-05-12 00:50:11,Spot,Transaction Related,DASH,-0.27000000,""
2021-05-12 00:50:11,Spot,Buy,BTC,0.00201636,""
2021-05-12 01:00:24,USDT-Futures,Referrer rebates,BNB,2.9E-7,""
2021-05-12 01:11:31,Spot,Fee,BNB,-0.00103110,""
2021-05-12 01:11:31,Spot,Transaction Related,DOT,-23.50000000,""
2021-05-12 01:11:31,Spot,Buy,BTC,0.01615014,""
2021-05-12 01:15:50,Spot,Fee,BNB,-0.00059265,""
2021-05-12 01:15:50,Spot,Transaction Related,DOTUP,-5.15000000,""
2021-05-12 01:15:50,Spot,Buy,USDT,530.69205000,""
2021-05-12 01:16:28,Spot,Fee,BNB,-0.00046095,""
2021-05-12 01:16:28,Spot,Transaction Related,DOTUP,-4.00000000,""
2021-05-12 01:16:28,Spot,Buy,USDT,412.18800000,""
2021-05-12 01:16:28,Spot,Buy,USDT,201.97212000,""
2021-05-12 01:16:28,Spot,Transaction Related,DOTUP,-1.96000000,""
2021-05-12 01:16:28,Spot,Fee,BNB,-0.00022586,""
2021-05-12 01:30:30,USDT-Futures,Referrer rebates,BNB,2.9E-7,""
2021-05-12 01:46:14,USDT-Futures,Referrer rebates,BNB,0.00000123,""
2021-05-12 02:20:08,USDT-Futures,Referrer rebates,BNB,0.00000190,""
2021-05-12 02:22:09,USDT-Futures,Referrer rebates,BNB,6.7E-7,""
2021-05-12 02:22:12,USDT-Futures,Referrer rebates,BNB,2.2E-7,""
2021-05-12 02:23:12,Spot,Transaction Related,USDT,-10.10161100,""
2021-05-12 02:23:12,Spot,Fee,BNB,-0.00001139,""
2021-05-12 02:23:12,Spot,Buy,DOTDOWN,12.01000000,""
2021-05-12 02:23:30,Spot,Buy,DOTDOWN,178.22000000,""
2021-05-12 02:23:30,Spot,Fee,BNB,-0.00016619,""
2021-05-12 02:23:30,Spot,Transaction Related,USDT,-149.90084200,""
2021-05-12 02:24:07,USDT-Futures,Referrer rebates,BNB,6.8E-7,""
2021-05-12 02:24:11,USDT-Futures,Referrer rebates,BNB,1.1E-7,""
2021-05-12 02:24:14,USDT-Futures,Referrer rebates,BNB,8.4E-7,""
2021-05-12 02:24:14,USDT-Futures,Referrer rebates,BNB,3.1E-7,""
2021-05-12 02:24:15,USDT-Futures,Referrer rebates,BNB,7.6E-7,""
2021-05-12 02:25:11,USDT-Futures,Referrer rebates,BNB,4.5E-7,""
2021-05-12 02:25:13,USDT-Futures,Referrer rebates,BNB,1.1E-7,""
2021-05-12 02:28:06,USDT-Futures,Referrer rebates,BNB,2.3E-7,""
2021-05-12 02:28:08,USDT-Futures,Referrer rebates,BNB,2.3E-7,""
2021-05-12 02:29:20,USDT-Futures,Referrer rebates,BNB,8.1E-7,""
2021-05-12 02:29:34,USDT-Futures,Referrer rebates,BNB,4.8E-7,""
2021-05-12 02:32:15,USDT-Futures,Referrer rebates,BNB,9.4E-7,""
2021-05-12 03:02:15,USDT-Futures,Referrer rebates,BNB,1.1E-7,""
2021-05-12 03:02:25,USDT-Futures,Referrer rebates,BNB,1.1E-7,""
2021-05-12 03:11:08,Spot,Buy,BNBDOWN,2.81000000,""
2021-05-12 03:11:08,Spot,Fee,BNB,-0.00002212,""
2021-05-12 03:11:08,Spot,Transaction Related,USDT,-20.00720000,""
2021-05-12 03:11:21,Spot,Buy,BNBDOWN,1.42000000,""
2021-05-12 03:11:21,Spot,Transaction Related,USDT,-10.11040000,""
2021-05-12 03:11:21,Spot,Fee,BNB,-0.00001117,""
2021-05-12 03:12:16,Spot,Fee,BNB,-0.00003174,""
2021-05-12 03:12:16,Spot,Buy,BNBDOWN,4.02000000,""
2021-05-12 03:12:16,Spot,Transaction Related,USDT,-28.62240000,""
2021-05-12 03:12:31,Spot,Fee,BNB,-0.00002361,""
2021-05-12 03:12:31,Spot,Transaction Related,USDT,-21.28880000,""
2021-05-12 03:12:31,Spot,Buy,BNBDOWN,2.99000000,""
2021-05-12 03:39:26,Spot,Transaction Related,USDT,-200.00008460,""
2021-05-12 03:39:26,Spot,Fee,BNB,-0.00022227,""
2021-05-12 03:39:26,Spot,Buy,BTCDOWN,4695.94000000,""
2021-05-12 03:45:14,Spot,Transaction Related,BTC,-0.00200018,""
2021-05-12 03:45:14,Spot,Fee,BNB,-0.00012735,""
2021-05-12 03:45:14,Spot,Buy,XEM,314.00000000,""
2021-05-12 03:45:42,Spot,Buy,OMG,9.80000000,""
2021-05-12 03:45:42,Spot,Fee,BNB,-0.00012743,""
2021-05-12 03:45:42,Spot,Transaction Related,BTC,-0.00200116,""
2021-05-12 03:49:15,USDT-Futures,Referrer rebates,BNB,1.1E-7,""
2021-05-12 03:49:15,USDT-Futures,Referrer rebates,BNB,2.3E-7,""
2021-05-12 03:50:12,USDT-Futures,Referrer rebates,BNB,2.3E-7,""
2021-05-12 03:50:15,USDT-Futures,Referrer rebates,BNB,0.00000253,""
2021-05-12 03:52:14,USDT-Futures,Referrer rebates,BNB,8.5E-7,""
2021-05-12 03:52:14,USDT-Futures,Referrer rebates,BNB,8.5E-7,""
2021-05-12 03:53:29,Spot,Buy,OMG,9.84000000,""
2021-05-12 03:53:29,Spot,Fee,BNB,-0.00012726,""
2021-05-12 03:53:29,Spot,Transaction Related,BTC,-0.00200047,""
2021-05-12 03:54:32,Spot,Buy,OMG,19.89000000,""
2021-05-12 03:54:32,Spot,Transaction Related,BTC,-0.00400186,""
2021-05-12 03:54:32,Spot,Fee,BNB,-0.00025685,""
2021-05-12 03:55:08,USDT-Futures,Referrer rebates,BNB,0.00000384,""
2021-05-12 03:56:39,Spot,Buy,BTC,0.00806016,""
2021-05-12 03:56:39,Spot,Transaction Related,OMG,-39.53000000,""
2021-05-12 03:56:39,Spot,Fee,BNB,-0.00051146,""
2021-05-12 03:56:43,Spot,Transaction Related,BTC,-0.00200025,""
2021-05-12 03:56:43,Spot,Buy,OMG,9.81000000,""
2021-05-12 03:56:43,Spot,Fee,BNB,-0.00012656,""
2021-05-12 04:00:22,USDT-Futures,Referrer rebates,BNB,0.00000145,""
2021-05-12 04:00:53,USDT-Futures,Referrer rebates,BNB,0.00000423,""
2021-05-12 04:05:08,USDT-Futures,Referrer rebates,BNB,4.4E-7,""
2021-05-12 04:05:08,USDT-Futures,Referrer rebates,BNB,0.00000110,""
2021-05-12 04:07:15,Spot,Fee,BNB,-0.00012812,""
2021-05-12 04:07:15,Spot,Buy,BTC,0.00201399,""
2021-05-12 04:07:15,Spot,Transaction Related,OMG,-9.81000000,""
2021-05-12 04:07:21,Spot,Buy,XTZ,16.64000000,""
2021-05-12 04:07:21,Spot,Transaction Related,BTC,-0.00200012,""
2021-05-12 04:07:21,Spot,Fee,BNB,-0.00012738,""
2021-05-12 04:13:03,Spot,Buy,XTZ,13.19000000,""
2021-05-12 04:13:03,Spot,Transaction Related,BTC,-0.00157752,""
2021-05-12 04:13:03,Spot,Transaction Related,BTC,-0.00042338,""
2021-05-12 04:13:03,Spot,Buy,XTZ,3.54000000,""
2021-05-12 04:13:03,Spot,Fee,BNB,-0.00002702,""
2021-05-12 04:13:03,Spot,Fee,BNB,-0.00010070,""
2021-05-12 04:18:05,Spot,Buy,XEM,316.00000000,""
2021-05-12 04:18:05,Spot,Transaction Related,BTC,-0.00200344,""
2021-05-12 04:18:05,Spot,Fee,BNB,-0.00012839,""
2021-05-12 04:43:11,USDT-Futures,Referrer rebates,BNB,6.2E-7,""
2021-05-12 04:50:46,Spot,Buy,BTC,0.00403109,""
2021-05-12 04:50:46,Spot,Fee,BNB,-0.00025618,""
2021-05-12 04:50:46,Spot,Transaction Related,XTZ,-33.37000000,""
2021-05-12 04:50:50,Spot,Transaction Related,BTC,-0.00202730,""
2021-05-12 04:50:50,Spot,Buy,IOTA,55.00000000,""
2021-05-12 04:50:50,Spot,Fee,BNB,-0.00012866,""
2021-05-12 04:52:53,Spot,Buy,IOTA,55.00000000,""
2021-05-12 04:52:53,Spot,Fee,BNB,-0.00012882,""
2021-05-12 04:52:53,Spot,Transaction Related,BTC,-0.00201850,""
2021-05-12 04:55:16,Spot,Transaction Related,USDT,-74.85910000,""
2021-05-12 04:55:16,Spot,Buy,EOSUP,6.70000000,""
2021-05-12 04:55:16,Spot,Fee,BNB,-0.00008268,""
2021-05-12 05:00:00,Spot,Transaction Related,EOSUP,-2.26000000,""
2021-05-12 05:00:00,Spot,Buy,USDT,25.67586000,""
2021-05-12 05:00:00,Spot,Fee,BNB,-0.00002842,""
2021-05-12 05:00:01,Spot,Fee,BNB,-0.00005585,""
2021-05-12 05:00:01,Spot,Buy,USDT,50.44284000,""
2021-05-12 05:00:01,Spot,Transaction Related,EOSUP,-4.44000000,""
2021-05-12 05:17:12,Spot,Buy,BTC,0.00407330,""
2021-05-12 05:17:12,Spot,Transaction Related,IOTA,-110.00000000,""
2021-05-12 05:17:12,Spot,Fee,BNB,-0.00025894,""
2021-05-12 05:40:16,Spot,Buy,WAN,52.00000000,""
2021-05-12 05:40:16,Spot,Fee,BNB,-0.00012804,""
2021-05-12 05:40:16,Spot,Transaction Related,BTC,-0.00202280,""
2021-05-12 05:59:08,Spot,Buy,WAN,9.00000000,""
2021-05-12 05:59:08,Spot,Fee,BNB,-0.00002214,""
2021-05-12 05:59:08,Spot,Transaction Related,BTC,-0.00034857,""
2021-05-12 06:04:09,USDT-Futures,Referrer rebates,BNB,2.3E-7,""
2021-05-12 06:04:11,USDT-Futures,Referrer rebates,BNB,4.6E-7,""
2021-05-12 06:09:14,USDT-Futures,Referrer rebates,BNB,8.2E-7,""
2021-05-12 06:15:52,Spot,Fee,BNB,-0.00010612,""
2021-05-12 06:15:52,Spot,Buy,WAN,43.00000000,""
2021-05-12 06:15:52,Spot,Transaction Related,BTC,-0.00166539,""
2021-05-12 06:17:12,USDT-Futures,Referrer rebates,BNB,4.1E-7,""
2021-05-12 06:23:13,USDT-Futures,Referrer rebates,BNB,0.00000125,""
2021-05-12 06:25:10,USDT-Futures,Referrer rebates,BNB,4.1E-7,""
2021-05-12 06:25:12,USDT-Futures,Referrer rebates,BNB,4.1E-7,""
2021-05-12 06:25:17,Spot,Fee,BNB,-0.00008367,""
2021-05-12 06:25:17,Spot,Transaction Related,USDT,-75.00060000,""
2021-05-12 06:25:17,Spot,Buy,XTZUP,77.40000000,""
2021-05-12 06:25:41,Spot,Buy,LTCUP,0.92000000,""
2021-05-12 06:25:41,Spot,Buy,LTCUP,0.43000000,""
2021-05-12 06:25:41,Spot,Fee,BNB,-0.00002644,""
2021-05-12 06:25:41,Spot,Transaction Related,USDT,-23.90198000,""
2021-05-12 06:25:41,Spot,Fee,BNB,-0.00005658,""
2021-05-12 06:25:41,Spot,Transaction Related,USDT,-51.13728000,""
2021-05-12 06:26:13,Spot,Buy,LINKUP,1.58000000,""
2021-05-12 06:26:13,Spot,Transaction Related,USDT,-41.19744000,""
2021-05-12 06:26:13,Spot,Fee,BNB,-0.00004586,""
2021-05-12 06:26:13,Spot,Transaction Related,USDT,-33.90364000,""
2021-05-12 06:26:13,Spot,Fee,BNB,-0.00003774,""
2021-05-12 06:26:13,Spot,Buy,LINKUP,1.92000000,""
2021-05-12 06:32:15,Spot,Fee,BNB,-0.00008468,""
2021-05-12 06:32:15,Spot,Transaction Related,XTZUP,-77.40000000,""
2021-05-12 06:32:15,Spot,Buy,USDT,76.31640000,""
2021-05-12 06:35:19,Spot,Fee,BNB,-0.00015837,""
2021-05-12 06:35:19,Spot,Buy,USDT,142.63680000,""
2021-05-12 06:35:19,Spot,Transaction Related,BNBDOWN,-19.38000000,""
2021-05-12 06:45:49,Spot,Fee,BNB,-0.00016780,""
2021-05-12 06:45:49,Spot,Transaction Related,USDT,-150.12734000,""
2021-05-12 06:45:49,Spot,Buy,LTCUP,2.74000000,""
2021-05-12 06:52:09,Spot,Transaction Related,BTC,-0.00402570,""
2021-05-12 06:52:09,Spot,Buy,WAN,105.00000000,""
2021-05-12 06:52:09,Spot,Fee,BNB,-0.00025979,""
2021-05-12 06:58:08,USDT-Futures,Referrer rebates,BNB,4.1E-7,""
2021-05-12 06:59:08,USDT-Futures,Referrer rebates,BNB,4.1E-7,""
2021-05-12 07:00:22,USDT-Futures,Referrer rebates,BNB,4.1E-7,""
2021-05-12 07:20:08,USDT-Futures,Referrer rebates,BNB,4.1E-7,""
2021-05-12 07:37:48,Spot,Fee,BNB,-0.00001530,""
2021-05-12 07:37:48,Spot,Buy,BTC,0.00024320,""
2021-05-12 07:37:48,Spot,Transaction Related,XEM,-38.00000000,""
2021-05-12 07:38:41,Spot,Transaction Related,XEM,-592.00000000,""
2021-05-12 07:38:41,Spot,Fee,BNB,-0.00024178,""
2021-05-12 07:38:41,Spot,Buy,BTC,0.00378880,""
2021-05-12 07:38:45,Spot,Fee,BNB,-0.00001279,""
2021-05-12 07:38:45,Spot,Transaction Related,BTC,-0.00017937,""
2021-05-12 07:38:45,Spot,Buy,DASH,0.02800000,""
2021-05-12 07:38:45,Spot,Fee,BNB,-0.00001142,""
2021-05-12 07:38:45,Spot,Transaction Related,BTC,-0.00020090,""
2021-05-12 07:38:45,Spot,Buy,DASH,0.02500000,""
2021-05-12 07:38:47,Spot,Transaction Related,BTC,-0.00056682,""
2021-05-12 07:38:47,Spot,Fee,BNB,-0.00003610,""
2021-05-12 07:38:47,Spot,Buy,DASH,0.07900000,""
2021-05-12 07:38:48,Spot,Buy,DASH,0.02500000,""
2021-05-12 07:38:48,Spot,Fee,BNB,-0.00001142,""
2021-05-12 07:38:48,Spot,Transaction Related,BTC,-0.00017937,""
2021-05-12 07:38:49,Spot,Buy,DASH,0.02600000,""
2021-05-12 07:38:49,Spot,Fee,BNB,-0.00001188,""
2021-05-12 07:38:49,Spot,Transaction Related,BTC,-0.00018655,""
2021-05-12 07:39:38,Spot,Transaction Related,BTC,-0.00020140,""
2021-05-12 07:39:38,Spot,Transaction Related,BTC,-0.00048912,""
2021-05-12 07:39:38,Spot,Fee,BNB,-0.00003115,""
2021-05-12 07:39:38,Spot,Fee,BNB,-0.00001282,""
2021-05-12 07:39:38,Spot,Buy,DASH,0.06800000,""
2021-05-12 07:39:38,Spot,Buy,DASH,0.02800000,""
2021-05-12 07:43:58,Spot,Fee,BNB,-0.00012873,""
2021-05-12 07:43:58,Spot,Transaction Related,DASH,-0.27900000,""
2021-05-12 07:43:58,Spot,Buy,BTC,0.00201717,""
2021-05-12 07:55:15,Spot,Buy,NEO,1.02000000,""
2021-05-12 07:55:15,Spot,Transaction Related,BTC,-0.00200634,""
2021-05-12 07:55:15,Spot,Fee,BNB,-0.00012816,""
2021-05-12 07:56:18,Spot,Fee,BNB,-0.00025567,""
2021-05-12 07:56:18,Spot,Buy,USDT,228.94593000,""
2021-05-12 07:56:18,Spot,Transaction Related,LTCUP,-4.09000000,""
2021-05-12 07:56:44,Spot,Fee,BNB,-0.00012926,""
2021-05-12 07:56:44,Spot,Buy,NEO,1.03000000,""
2021-05-12 07:56:44,Spot,Transaction Related,BTC,-0.00201674,""
2021-05-12 08:00:11,Spot,Buy,XTZDOWN,338.52000000,""
2021-05-12 08:00:11,Spot,Transaction Related,USDT,-1.86863040,""
2021-05-12 08:00:11,Spot,Fee,BNB,-0.00002058,""
2021-05-12 08:00:11,Spot,Buy,XTZDOWN,3284.67000000,""
2021-05-12 08:00:11,Spot,Transaction Related,USDT,-18.13137840,""
2021-05-12 08:00:11,Spot,Fee,BNB,-0.00000212,""
2021-05-12 08:00:25,Spot,Fee,BNB,-0.00004516,""
2021-05-12 08:00:25,Spot,Transaction Related,USDT,-40.00004800,""

[...]

2021-06-24 03:36:09,Spot,Fee,BNB,-0.00013027,""
2021-06-24 03:36:09,Spot,Buy,USDT,-49.98490800,""
2021-06-24 03:36:09,Spot,Buy,BTCDOWN,764.88000000,""
2021-06-24 03:42:24,Spot,Fee,BNB,-0.00013300,""
2021-06-24 03:42:24,Spot,Sell,USDT,50.82627600,""
2021-06-24 03:42:24,Spot,Sell,BTCDOWN,-764.88000000,""
2021-06-24 04:09:09,Spot,Buy,BTCDOWN,758.27000000,""
2021-06-24 04:09:09,Spot,Buy,USDT,-49.96241030,""
2021-06-24 04:09:09,Spot,Fee,BNB,-0.00013052,""
2021-06-24 04:21:07,Spot,Sell,USDT,50.80409000,""
2021-06-24 04:21:07,Spot,Sell,BTCDOWN,-758.27000000,""
2021-06-24 04:21:07,Spot,Fee,BNB,-0.00013408,""
2021-06-24 05:01:07,USDT-Futures,Referrer rebates,BNB,3.3E-7,""
2021-06-24 05:39:08,Spot,Fee,BNB,-0.00013040,""
2021-06-24 05:39:08,Spot,Buy,BTCDOWN,761.74000000,""
2021-06-24 05:39:08,Spot,Buy,USDT,-50.00061360,""
2021-06-24 05:50:33,Spot,Buy,USDT,-98.81178100,""
2021-06-24 05:50:33,Spot,Fee,BNB,-0.00000312,""
2021-06-24 05:50:33,Spot,Buy,BTCDOWN,18.37000000,""
2021-06-24 05:50:33,Spot,Buy,BTCDOWN,1527.23000000,""
2021-06-24 05:50:33,Spot,Buy,USDT,-1.18853900,""
2021-06-24 05:50:33,Spot,Fee,BNB,-0.00026043,""
2021-06-24 06:47:07,USDT-Futures,Referrer rebates,BNB,2.5E-7,""
2021-06-24 06:47:07,USDT-Futures,Referrer rebates,BNB,0.00000738,""
2021-06-24 06:50:09,USDT-Futures,Referrer rebates,BNB,1.9E-7,""
2021-06-24 06:50:09,USDT-Futures,Referrer rebates,BNB,9E-8,""
2021-06-24 06:50:21,USDT-Futures,Referrer rebates,BNB,0.00000144,""
2021-06-24 06:56:20,USDT-Futures,Referrer rebates,BNB,1.3E-7,""
2021-06-24 07:15:41,USDT-Futures,Referrer rebates,BNB,0.00000126,""
2021-06-24 07:25:10,USDT-Futures,Referrer rebates,BNB,0.00000144,""
2021-06-24 07:25:10,USDT-Futures,Referrer rebates,BNB,3.6E-7,""
2021-06-24 07:30:13,USDT-Futures,Referrer rebates,BNB,1.0E-7,""
2021-06-24 07:31:18,USDT-Futures,Referrer rebates,BNB,1.8E-7,""
2021-06-24 07:31:18,USDT-Futures,Referrer rebates,BNB,7.2E-7,""
2021-06-24 08:19:19,USDT-Futures,Referrer rebates,BNB,1.5E-7,""
2021-06-24 08:19:39,USDT-Futures,Referrer rebates,BNB,1.0E-7,""
2021-06-24 08:22:07,USDT-Futures,Referrer rebates,BNB,0.00000144,""
2021-06-24 08:54:53,Spot,Buy,USDT,-200.00017710,""
2021-06-24 08:54:53,Spot,Fee,BNB,-0.00049557,""
2021-06-24 08:54:53,Spot,Buy,BTCDOWN,3188.27000000,""
2021-06-24 09:12:08,USDT-Futures,Referrer rebates,BNB,8E-8,""
2021-06-24 09:41:18,USDT-Futures,Referrer rebates,BNB,7.2E-7,""
2021-06-24 09:44:41,Spot,Commission History,BNB,0.00002391,""
2021-06-24 09:45:27,Spot,Commission History,BNB,0.00002250,""
2021-06-24 09:45:55,Spot,Commission History,BNB,0.00000311,""
2021-06-24 09:45:55,Spot,Commission History,BNB,0.00001770,""
2021-06-24 09:46:25,Spot,Commission History,BNB,0.00002100,""
2021-06-24 11:09:52,USDT-Futures,Referrer rebates,BNB,3E-8,""
2021-06-24 11:11:20,USDT-Futures,Referrer rebates,BNB,0.00000343,""
2021-06-24 12:04:16,USDT-Futures,Referrer rebates,BNB,3.6E-7,""
2021-06-24 12:04:25,Spot,Fee,BNB,-0.00098766,""
2021-06-24 12:04:25,Spot,Buy,BTCDOWN,6803.88000000,""
2021-06-24 12:04:25,Spot,Buy,USDT,-400.00010520,""
2021-06-24 12:31:30,USDT-Futures,Referrer rebates,BNB,1.2E-7,""
2021-06-24 12:48:08,USDT-Futures,Referrer rebates,BNB,1.2E-7,""
2021-06-24 12:48:19,USDT-Futures,Referrer rebates,BNB,9E-8,""
2021-06-24 13:20:20,USDT-Futures,Referrer rebates,BNB,3.2E-7,""
2021-06-24 13:22:55,USDT-Futures,Referrer rebates,BNB,3.2E-7,""
2021-06-24 15:13:09,USDT-Futures,Referrer rebates,BNB,5E-8,""
2021-06-24 15:13:15,USDT-Futures,Referrer rebates,BNB,3.6E-7,""
2021-06-24 15:24:54,Spot,Fee,BNB,-0.00196800,""
2021-06-24 15:24:54,Spot,Buy,BTCDOWN,13956.73000000,""
2021-06-24 15:24:54,Spot,Buy,USDT,-799.99976360,""
2021-06-24 15:53:30,Spot,Sell,BTCDOWN,-401.85000000,""
2021-06-24 15:53:30,Spot,Fee,BNB,-0.00375564,""
2021-06-24 15:53:30,Spot,Sell,USDT,23.88596400,""
2021-06-24 15:53:30,Spot,Sell,BTCDOWN,-25854.37000000,""
2021-06-24 15:53:30,Spot,Sell,USDT,1536.78375280,""
2021-06-24 15:53:30,Spot,Fee,BNB,-0.00005837,""
2021-06-24 16:39:09,Spot,Buy,BTCDOWN,859.85000000,""
2021-06-24 16:39:09,Spot,Fee,BNB,-0.00012138,""
2021-06-24 16:39:09,Spot,Buy,USDT,-50.00027750,""
2021-06-24 16:52:49,Spot,Buy,BTCDOWN,1744.60000000,""
2021-06-24 16:52:49,Spot,Fee,BNB,-0.00024266,""
2021-06-24 16:52:49,Spot,Buy,USDT,-100.00047200,""
2021-06-24 16:58:07,USDT-Futures,Referrer rebates,BNB,7.2E-7,""
2021-06-24 17:05:30,Spot,Buy,BTCDOWN,3599.07000000,""
2021-06-24 17:05:30,Spot,Fee,BNB,-0.00048476,""
2021-06-24 17:05:30,Spot,Buy,USDT,-200.00031990,""
2021-06-24 18:02:28,USDT-Futures,Referrer rebates,BNB,0.00000310,""
2021-06-24 19:58:11,USDT-Futures,Referrer rebates,BNB,3E-8,""
2021-06-24 20:57:09,USDT-Futures,Referrer rebates,BNB,3.1E-7,""
2021-06-24 22:18:21,USDT-Futures,Referrer rebates,BNB,3.1E-7,""
2021-06-24 22:59:09,USDT-Futures,Referrer rebates,BNB,3E-8,""
2021-06-24 23:23:38,USDT-Futures,Referrer rebates,BNB,0.00000284,""
2021-06-24 23:25:09,USDT-Futures,Referrer rebates,BNB,0.00000260,""
2021-06-24 23:26:15,USDT-Futures,Referrer rebates,BNB,3E-8,""
2021-06-25 00:02:07,USDT-Futures,Referrer rebates,BNB,4E-8,""
2021-06-25 00:06:12,USDT-Futures,Referrer rebates,BNB,5E-8,""
2021-06-25 00:10:31,USDT-Futures,Referrer rebates,BNB,0.00000230,""
2021-06-25 00:13:14,USDT-Futures,Referrer rebates,BNB,4E-8,""
2021-06-25 00:26:07,USDT-Futures,Referrer rebates,BNB,1.8E-7,""
2021-06-25 00:56:07,USDT-Futures,Referrer rebates,BNB,3.2E-7,""
2021-06-25 00:57:36,USDT-Futures,Referrer rebates,BNB,3.2E-7,""
2021-06-25 01:01:37,Spot,Fee,BNB,-0.00086598,""
2021-06-25 01:01:37,Spot,Sell,USDT,355.89594240,""
2021-06-25 01:01:37,Spot,Sell,BTCDOWN,-6203.52000000,""
2021-06-25 01:05:04,USDT-Futures,Referrer rebates,BNB,0.00000108,""
2021-06-25 01:58:14,USDT-Futures,Referrer rebates,BNB,1E-8,""
2021-06-25 02:00:10,Spot,Fee,BNB,-0.00012291,""
2021-06-25 02:00:10,Spot,Buy,USDT,-50.00035320,""
2021-06-25 02:00:10,Spot,Buy,BTCDOWN,904.82000000,""

--- a/src/book.py
+++ b/src/book.py
@@ -75,6 +75,7 @@ class Book:
             "Savings Principal redemption": "CoinLendEnd",
             "Commission History": "Commission",
             "Commission Fee Shared With You": "Commission",
+            "Referrer rebates": "Commission",
             "Launchpool Interest": "StakingInterest",
             "Cash Voucher distribution": "Airdrop",
             "Super BNB Mining": "StakingInterest",
@@ -108,6 +109,12 @@ class Book:
                     "Buy",
                 ):
                     operation = "Sell" if change < 0 else "Buy"
+
+                if operation in ("Commission") and account != "Spot":
+                    # All comissions will be handled the same way.
+                    # As of now, only Spot Binance Operations are supported,
+                    # so we have to change the account type to Spot.
+                    account = "Spot"
 
                 change = abs(change)
 

